content/swift/concepts/protocols/terms/decode/decode.md

```diff
@@ -0,0 +1,93 @@
+---
+Title: 'Decoder'
+Description: 'Enables values to be decoded from a native format to a custom format.'
+Subjects:
+  - 'iOS'
+  - 'Mobile Development'
+Tags:
+  - 'Protocols'
+Catalog Content:
+  - 'learn-swift'
+  - 'paths/ios-developer'
+---
+
+**`Decoder`** is a type that allows the decoding of values from a native format such as JSON or XML  into an in-memory representation like a struct, class, or enum.
+
+> **Note:** If you are also looking to encode values into a native representation, it would be beneficial to use the `Codable` protocol as it supports the use of both `Encodable` and `Decodable`.
+
+
+## Syntax
+In order to `Decode` values, they must conform to the `Decodable` protocol as it initializes from a decoder
+```pseudo 
+struct someStruct: Decodable {
+	// Properties that conform to Decodable
+}
+```
+In the syntax above, all struct properties will be decoded
+
+## Example
+In this example a `Dog`
+`struct` is declared as well as a `dogData` JSON string. It is then converted from JSON to a `Dog` object.
+```swift
+import Foundation
+
+struct Dog : Decodable {
+  let name: String
+  let age: Int
+  let breed: String
+}
+
+// JSON string to be decoded
+let dogData = """
+{
+  "name": "Clifford",
+  "age": 10,
+  "breed": "Big Red"
+}
+""".data(using: .utf8)!
+
+// Converting the JSON String to a Dog object
+let decoder = JSONDecoder()
+let dogString = try decoder.decode(Dog.self, from: dogData)
+print(dogString)
+```
+
+This will output `dogData` as a `Dog` object:
+```shell
+Dog(name: "Clifford", age: 10, breed: "Big Red")
+```
+
+## Example using Codable
+As mentioned above, using the `Codable` protocol over `Decodable` or `Encodable` is beneficial in scenarios in which you want to both `Encode` and `Decode` objects. 
+
+In the below example a `Car`
```

Comment:
```suggestion
In the below example a `Car` struct is declared and an instance `car` is created. It's data is encoded to a JSON string, then decoded back to a Swift object.
```

Reply:
code formatting should be used when referencing a specific element within the code, such as a variable, or a specific thing/term in the given language. In this circumstance it seems like encoded and decoded are used generically. It is not referring to something specific such as the variable `decoder` or the protocol `Codable`